scripts/test/system.sh

```diff
@@ -68,6 +68,35 @@ test_subctl_gather
 _subctl diagnose all --validation-timeout 20
 _subctl diagnose firewall inter-cluster --validation-timeout 20 "${KUBECONFIGS_DIR}"/kind-config-cluster1 "${KUBECONFIGS_DIR}"/kind-config-cluster2
 
+# Test subctl diagnose in-cluster
+cat <<EOF | kubectl apply -f -
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: submariner-diagnose
+  namespace: submariner-operator
+spec:
+  template:
+    metadata:
+      labels:
+        submariner.io/transient: "true"
+    spec:
+      containers:
+      - name: submariner-diagnose
+        image: quay.io/submariner/subctl:devel
+        command: ["subctl",  "diagnose", "all", "--in-cluster"]
+      restartPolicy: Never
+      serviceAccount: submariner-diagnose
+      serviceAccountName: submariner-diagnose
+  backoffLimit: 0
+EOF
+kubectl wait --for=condition=complete job/submariner-diagnose -n submariner-operator --timeout 120s
+retVal=$?
+if [ $retVal -ne 0 ]; then
```

Comment:
```suggestion

kubectl wait --for=condition=complete job/submariner-diagnose -n "$subm_ns"  --timeout 120s &
kubectl wait --for=condition=failed job/submariner-diagnose -n "$subm_ns"  --timeout 120s && exit 1 &
if ! wait -n; then
    kill %1 %2 2> /dev/null
```

This avoids waiting for 2 minutes for it to time out in case of failure.